tests/php/View/SSViewerTest.php

```diff
@@ -552,6 +552,25 @@ public function testCurrentScopeLoopWith()
         $this->assertEquals("SubKid1SubKid2Number6", $result, "Loop in current scope works");
     }
 
+    public function testArgumentTypes()
+    {
+        $this->assertEquals(
+            'arg1:0,arg2:"string",arg3:true' . "\n" .
+            'arg1:false,arg2:"string",arg3:true' . "\n" .
+            'arg1:null,arg2:"string",arg3:true' . "\n" .
+            'arg1:"",arg2:"string",arg3:true' . "\n" .
+            'arg1:0,arg2:1,arg3:2',
+            $this->render(
+                '$methodWithTypedArguments(0, "string", true).RAW' . "\n" .
+                '$methodWithTypedArguments(false, "string", true).RAW' . "\n" .
+                '$methodWithTypedArguments(null, "string", true).RAW' . "\n" .
+                '$methodWithTypedArguments("", "string", true).RAW' . "\n" .
+                '$methodWithTypedArguments(0, 1, 2).RAW',
+                new TestViewableData()
+            )
+        );
+    }
```

Comment:
```suggestion
    public function provideArgumentTypes()
    {
        return [
            [
                'arg1:0,arg2:"string",arg3:true',
                '$methodWithTypedArguments(0, "string", true).RAW',
            ],
            [
                'arg1:false,arg2:"string",arg3:true'.
                '$methodWithTypedArguments(false, "string", true).RAW',
            ],
            [
                'arg1:null,arg2:"string",arg3:true',
                '$methodWithTypedArguments(null, "string", true).RAW',
            ],
            [
                'arg1:"",arg2:"string",arg3:true',
                '$methodWithTypedArguments("", "string", true).RAW',
            ],
            [
                'arg1:0,arg2:1,arg3:2',
                '$methodWithTypedArguments(0, 1, 2).RAW',
            ],
        ];
    }

    /**
     * @dataProvider provideArgumentTypes
     */
    public function testArgumentTypes(string $expected, string $template)
    {
        $this->assertEquals($expected, $this->render($template, new TestViewableData()));
    }
```

Please use a data provider so it's easier to debug if something does fail.

Reply:
Done.

But there are many other test in the same file which does the same, maybe they should be rewritten too.

Reply:
They probably should be, but that's outside the scope of this pr and is a very low priority.
It makes sense to do new code in a better way than the existing code, where appropriate.